### ISSUE TITLE
how to flash it?

### ISSUE THREAD

**Issue by lyrics1988123** [spleenware/ripple#14]
guys im just too stupid. how can i use the commandline code?
i need the following but when i try to run it, my cmd tells me, that the command "~" is wrong or not existant (windows).. never flashed an esp32 before.. please help >.<

~/Library/Arduino15/packages/esp32/tools/esptool_py/2.6.1/esptool --chip esp32 --port /dev/cu.SLAB_USBtoUART --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xe000 ~/Library/Arduino15/packages/esp32/hardware/esp32/1.0.4/tools/partitions/boot_app0.bin 0x1000 ~/Library/Arduino15/packages/esp32/hardware/esp32/1.0.4/tools/sdk/bin/bootloader_dio_80m.bin 0x10000 Ripple-BLE.TTGO-V1.bin 0x8000 Ripple-BLE.TTGO-V1.partitions.bin

**Comment by spleenware** [comment]
Hi,
The '~' just means the user's home directory. For Windows you need to find where the esptool.exe is (which I think you have figured out). With the error above, change the '/' to a '\' before esptool

**Comment by spleenware** [comment]
ah, this editor masked the 'backslash' I typed in. Change the forward slash to a backslash before esptool.exe

**Comment by lyrics1988123** [comment]
hey, thanks for the very fast reply!
it was a typo.. oh lol
but i still get the same issue:

```
C:\Users\waiss\Downloads\lora>C:\Users\waiss\AppData\Local\Arduino15\packages\esp32\tools\esptool_py\2.6.1\esptool.exe --chip esp32 --port COM4 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xe000 C:\Users\waiss\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4\tools\partitions\boot_app0.bin 0x1000 C:\Users\waiss\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4\tools\sdk\bin\bootloader_dio_80m.bin 0x10000 Ripple-BLE.Heltech-v2.bin 0x8000 Ripple-BLE.Heltech-v2.partitions.bin
usage: esptool write_flash [-h] [--erase-all]
                           [--flash_freq {keep,40m,26m,20m,80m}]
                           [--flash_mode {keep,qio,qout,dio,dout}]
                           [--flash_size FLASH_SIZE]
                           [--spi-connection SPI_CONNECTION] [--no-progress]
                           [--verify] [--compress | --no-compress]
                           <address> <filename> [<address> <filename> ...]
esptool write_flash: error: argument <address> <filename>: Detected overlap at address: 0xe000 for file: C:\Users\waiss\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4\tools\partitions\boot_app0.bin

C:\Users\waiss\Downloads\lora>
```

**Comment by lyrics1988123** [comment]
tried it with my 2nd board, same issue.

**Comment by dwhacks** [comment]
Try using the ESP loader programme directly available from esp. I found it to be the easiest way on windows.

**Comment by lyrics1988123** [comment]
tried that.. and it looks like its flashing but i still have the standard programm running on it (a wifi test sketch).. =(

**Comment by lyrics1988123** [comment]
any idea? really wanted to try it out =(

**Comment by spleenware** [comment]
> any idea? really wanted to try it out =(

I have done a build on a Windows machine, for the Heltec V2 target, and captured the install commands and BIN files. Hopefully these work on your machine. Download this zip: https://drive.google.com/open?id=1dh9NOvUL0unOFCqyFuoMIfoMbTZG_yHT

**Comment by spleenware** [comment]
> any idea? really wanted to try it out =(

Also, I made two builds, the USB one and Bluetooth Classic.

**Comment by lyrics1988123** [comment]
thought this board has BL "LE" like you said in the other thread? x)
but thank you very much! will try it in the evening!
will let you know

**Comment by spleenware** [comment]
Yes, it has BLE and Classic, but the Classic has been more reliable in my testing. I 'thought' that I had the BLE stuff working well, but I'm still getting problems with it :-(

**Comment by lyrics1988123** [comment]
flashing did work (but the displays shows nothing if it should..)

i was able to connect to it with ripple android app over BT, also i created a group with the scan function. 
but.. when i try to send a message it says "unable to send message" (on both smartphones)

**Comment by spleenware** [comment]
Make sure both indicators on the toolbar are solid, ie. solid white bluetooth icon, and solid white 'wifi' symbol to the right. One is whether the handset has established connection to its own Ripple device, the other is if the other ripple device is reachable, and what the signal strength is.

**Comment by lyrics1988123** [comment]
hm.. looks like an issue with my smartphone maybe.. cause with my "old" CAT S50 the BT connection works (classic) and with my relative new huawai mate 20 lite it does not. it can find the ripple device and  i thought it does connect to it, but the icons on the top stays the same. any idea what could cause this?

**Comment by spleenware** [comment]
With Classic, make sure you do the BT pairing first, then do the CHOOSE DEVICE in the preferences screen. Also, as each device will have the same name ('Ripple Device'), make sure only one is turned ON when you are doing the pairing/preferences.

**Comment by lyrics1988123** [comment]
thank you! that was the issue, it works now :D

**Comment by spleenware** [comment]
> thank you! that was the issue, it works now :D

AH, good to hear. I probably should prepare a 'trouble shooting' Wiki page or something.

**Comment by lyrics1988123** [comment]
yeah the issue is, that all of your informations are spreaded in github and on instructables or where it was.. a bit confusing haha

but thanks for your help, was just testing it and i love it.
